validator/testcases/javascript/predefinedentities.py

```diff
@@ -575,32 +689,72 @@
                                    "dangerous and requires careful review "
                                    "by an administrative reviewer.",
                         "editors_only": True,
+                        "signing_severity": "high",
                       }},
                       u"nsIDOMGeoGeolocation": {"dangerous":
                          "Use of the geolocation API by add-ons requires "
                          "prompting users for consent."},
-                      u"nsIX509CertDB": {"dangerous": {
-                          "description": "Access to the X509 certificate "
-                                         "database is potentially dangerous "
-                                         "and requires careful review by an "
-                                         "administrative reviewer.",
+                      u"nsIWindowsRegKey": {"dangerous": {
+                          "signing_severity": "low",
                           "editors_only": True,
-                      }}}
+                          "description": (
+                              "Access to the registry is potentially "
+                              "dangerous, and should be reviewed with special "
+                              "care.")}},
+                      }
+
+CONTRACT_ENTITIES = {}
+
+DANGEROUS_CERT_DB = {
+  "description": "Access to the X509 certificate "
+                 "database is potentially dangerous "
+                 "and requires careful review by an "
+                 "administrative reviewer.",
+  "editors_only": True,
+  "signing_severity": "high",
+}
+
+INTERFACE_ENTITIES.update(
+    (interface, {"dangerous": DANGEROUS_CERT_DB})
+    for interface in ("nsIX509CertDB", "nsIX509CertDB2", "nsIX509CertList",
+                      "nsICertOverrideService"))
+
+CONTRACT_ENTITIES.update(
+    (contract, DANGEROUS_CERT_DB)
+    for contract in ("@mozilla.org/security/x509certdb;1",
+                     "@mozilla.org/security/x509certlist;1",
+                     "@mozilla.org/security/certoverride;1"))
+
 for interface in INTERFACES:
     def construct(interface):
         def wrap():
             return INTERFACES[interface]
         return wrap
-    INTERFACE_ENTITIES[interface] = {"xpcom_map": construct(interface)}
+    if interface not in INTERFACE_ENTITIES:
+        INTERFACE_ENTITIES[interface] = {}
+    INTERFACE_ENTITIES[interface]["xpcom_map"] = construct(interface)
 
 
 def build_quick_xpcom(method, interface, traverser, wrapper=False):
     """A shortcut to quickly build XPCOM objects on the fly."""
+    extra = ()
```

Comment:
On Tue, May 05, 2015 at 09:00:04AM -0700, Mathieu Agopian wrote:
>>
>>
>>  def build_quick_xpcom(method, interface, traverser, wrapper=False):
>>      """A shortcut to quickly build XPCOM objects on the fly."""
>> +    extra = ()
>
>I didn't see a test for those changes, did I miss it?

There really aren't any tests for most of the validator 
internals. I'm not even sure where I'd put it. I did add tests 
for the services in `Services` object (for which I implemented 
this), which check that the behavior is the same whether the 
services is instantiated in JS (as above), or accessed via the 
services object, so this is being tested indirectly.